### PR TITLE
Added Guard definition (ternary)

### PR DESCRIPTION
**Pull request by Swizz**
Hi,

Based on PR #32, this is my solution for guards through ternary operations.

This may raise some discuss about syntax linting and language hijack.

### CODE REVIEW

**Comment by hemanth** [comment]
LGTM 

tJeTfLOfl98Mo

What say @stoeffel?

**Comment by Swizz** [comment]
But.. using this way, the  `otherwise` clause is always required.

I was also thinking about a more tricky way using boolean short-circuit evaluation and truthy/falsy.
But this is truly not readable.

```js
const max = (a, b) => 
  (a > b) && a
  || /* otherwise */ b; 

const sign = (n) =>
  (n < 0) && -1 
  || (n >= 0) && 1 
```

**Comment by hemanth** [comment]
@Swizz heh heh bit twiddling is always fun, but readability goes to -Infinity!

File: readme.md [hunk 1]
```diff
@@ -203,9 +203,23 @@ Points-free function definitions look just like normal assignments without `func
 
 TODO
 
-## Guarded Functions
+## Guards
+Guards are boolean expressions that prevent evaluation of code unless the guard evaluates to true.
+It can be simulated thanks to the conditional ternary Operator.
```
Comment (jethrolarson) [inline readme.md]: You don't need ternary. And they're not simulated. They're just part of the js language.
e.g. `a && b && c` 
a guards the evaluation of b which guards the evaluation of c.

Comment (jethrolarson) [inline readme.md]: I might be wrong here though. Reading here http://c2.com/cgi/wiki?GuardClause gives me a different impression. Though the definition there may be more about procedural programming.

**Comment by Swizz** [comment]
@jethrolarson This is the definition of #32.

My diff on it is
```diff
+It can be simulated thanks to the conditional ternary Operator.
```

My english skills are very poor to attempt to write a real definition.
I gave you my coding solution to solve this issue and let you write something more suitable.

Cheer,

File: readme.md (hunk 1 above)
Comment (nickzuber) [inline readme.md]: I understand a guard as simply some boolean condition that prevents the guarded expression from being evaluated if that condition does not pass. A ternary or a simply if statement can be considered a guard in the right context, no? 

Chained `&&` and short circuiting also is a way to do that too in JS and many other languages, maybe it'd be a good idea to provide a few short examples of this?

**Comment by nickzuber** [comment]
I would love to see this get revised and merged

**Comment by jethrolarson** [comment]
Can reopen if revised

**Pull request closed without merging**